src/utils/registryExperience.ts

```diff
@@ -4,15 +4,25 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { AzureWizardPromptStep, ContextValueFilter, IActionContext, QuickPickWizardContext, RecursiveQuickPickStep, runQuickPickWizard } from '@microsoft/vscode-azext-utils';
-import * as vscode from 'vscode';
+import { CommonRegistryDataProvider } from '@microsoft/vscode-docker-registries';
+import { ext } from '../extensionVariables';
+import { UnifiedRegistryTreeDataProvider } from '../tree/registries/UnifiedRegistryTreeDataProvider';
+
+export async function registryExperience<TPick>(context: IActionContext, tdp: CommonRegistryDataProvider | CommonRegistryDataProvider[], contextValueFilter: ContextValueFilter, skipIfOne: boolean = true): Promise<TPick> {
+    let unifiedProvider: UnifiedRegistryTreeDataProvider | undefined;
+    if (Array.isArray(tdp)) {
+        unifiedProvider = new UnifiedRegistryTreeDataProvider(ext.context.globalState);
```

Comment:
I don't like the idea of instantiating more than one of these

Reply:
Gotcha, I agree with you then. I do want to find a way so that users don't get prompted by options that they can't act on, but we can potentially do that another way. But that's something we can worry about later on when we work on UX